packages/features/ee/impersonation/lib/ImpersonationProvider.ts

```diff
@@ -103,35 +107,39 @@ const ImpersonationProvider = CredentialsProvider({
     }
 
     // Check session
-    const sessionUserFromDb = await prisma.user.findUnique({
-      where: {
-        id: session?.user.id,
-      },
-      include: {
-        teams: {
-          where: {
-            AND: [
-              {
-                role: {
-                  in: ["ADMIN", "OWNER"],
+    if (teamId) {
```

Comment:
[Prefer Early Returns](https://handbook.cal.com/engineering/best-practices/prefer-early-returns) 

```suggestion
    if (!teamId) throw new Error("You do not have permission to do this.");
```